services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/events.py

```diff
@@ -238,10 +273,102 @@ async def action(cls, app: FastAPI, scheduler_data: SchedulerData) -> None:
         scheduler_data.dynamic_sidecar.was_compose_spec_submitted = True
 
 
+class RemoveUserCreatedServices(DynamicSchedulerEvent):
+    """
+    Triggered if the service is marked for remova.
+    The state of the service will be stored.
+    The dynamic-sidcar together with spawned containers
+    and dedicated network will be removed.
+    """
+
+    @classmethod
+    async def will_trigger(cls, app: FastAPI, scheduler_data: SchedulerData) -> bool:
+        return scheduler_data.dynamic_sidecar.service_removal_data.can_remove
+
+    @classmethod
+    async def action(cls, app: FastAPI, scheduler_data: SchedulerData) -> None:
+        # invoke container cleanup at this point
+        dynamic_sidecar_client: DynamicSidecarClient = get_dynamic_sidecar_client(app)
+
+        try:
+            await dynamic_sidecar_client.begin_service_destruction(
+                dynamic_sidecar_endpoint=scheduler_data.dynamic_sidecar.endpoint
+            )
+        except Exception as e:  # pylint: disable=broad-except
```

Comment:
I've updated log message to be more useful.

The idea here is, that if there is any kind of error while trying while contacting the  dynamic-sidecar, the removal procedure must finish.

Regarding what happens. If the dynamic-sidecar does not exist, it is expected for the service to get removed form the internal tracking of the scheduler.
If the service cannot be contacted for some reason a zombie will remain. But the likelihood of this should be very very low (I think the only network partitions will cause this).

Reply:
ok, so just maybe one thing. when the director is shutting down, there will be a cancellation error coming this way I gather. If you swallow it, then it will not proceed. so you might need to clean and then re-raise. or carefully check if it's actually ok to swallow it in this place.